From e1fb4edb18e56e1f6d10bd7bb32382b1a05c262a Mon Sep 17 00:00:00 2001
From: Alex Ermolov <aaermolov@gmail.com>
Date: Tue, 30 Dec 2025 14:48:59 +0300
Subject: [PATCH] Replace `InputDevice.fn` accesses with `InputDevice.path`

The `fn` attribute was deprecated in evdev 1.0.0 (Jun 02, 2018), and has now been removed.
---
 src/keyszer/devices.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/keyszer/devices.py b/src/keyszer/devices.py
index 767fd48..87b10fe 100644
--- a/src/keyszer/devices.py
+++ b/src/keyszer/devices.py
@@ -81,7 +81,7 @@ class DeviceRegistry:
             self.grab(device)
 
     def grab(self, device):
-        info(f"Grabbing '{device.name}' ({device.fn})", ctx="+K")
+        info(f"Grabbing '{device.name}' ({device.path})", ctx="+K")
         self._loop.add_reader(device, self._input_cb, device)
         self._devices.append(device)
         try:
@@ -104,7 +104,7 @@ class DeviceRegistry:
     def ungrab_by_filename(self, fn):
         for device in self._devices:
             try:
-                if device.fn == fn:
+                if device.path == fn:
                     info(f"Ungrabbing '{device.name}' (removed)", ctx="-K")
                     self._loop.remove_reader(device)
                     self._devices.remove(device)
@@ -143,7 +143,7 @@ class DeviceFilter:
         # picks up keyboard-ish devices.
         if self.matches:
             for match in self.matches:
-                if device.fn == match or device.name == match:
+                if device.path == match or device.name == match:
                     return True
             return False
 
-- 
2.51.2

